File: sourceCode/kotlin/ShapesVersionOne.kt
/**************************************************************************************************************
 * Printing the triangle both straightup and in the reverse directions...
 * @version 1.0
 * @author Coindsys
 * @date Thu, 09Jan2020
 **************************************************************************************************************
 */

 class Shapes {
 
    val numOfLines = 5;
    var numOfStars = 1;
    var margin = 10;
    //no constructor here as well, to keep things simple...
    
    /** Function to display a triangle...
     */
    fun displayTriangle() {

        for (i in 1..numOfLines) {
            for (k in 1..(numOfLines - i)) {
                print(" ");
            }
            for (j in 1..numOfStars) {
                print("*");
            }
            numOfStars += 2;
            println();
        }
    }
    
    fun displayInvertedTriangle() {
        numOfStars -= 2;
        for (i in 1..numOfLines) {
            for (k in 1..i) {
                print(" ");
            }
            for (j in 1..numOfStars) {
                print("*");
            }
            numOfStars -= 2;
            println();
        }
    }
 }

 fun main(args : Array<String>) {
    println("Hello Kotlin!!!");

    val myShapes = Shapes();
    myShapes.displayTriangle();
    myShapes.displayInvertedTriangle();
    readLine();

 }
 

File: sourceCode/c++/ShapesVersionOne.cpp
/**
 *
 */
 
#include <iostream>
#include <conio.h>
#include <stdio.h>
using namespace std;

class Shapes {
  
  public:
    //Again, no Constructor nothing, for the time being...
    void displayTriangle() {
      int numOfLines = 10;
      int margin = 20;
      int numOfStars = 1;
  
      for (int i = 0; i < numOfLines; i++) {
        for (int k = 0; k < numOfLines - i; k++) { 
          cout << " ";
        }
        for (int j=0; j < numOfStars; j++) {
          cout << "*";
        }
        numOfStars += 2;
        cout << endl;
      }
      getch();
    }
    
    void displayReversedTriangle() {
      int numOfLines = 10;
      int numOfStars = 19;
  
      for (int i = 0; i < numOfLines; i++) {
        for (int k = 0; k < i; k++) {
          //cout << " ";
          printf(" ");
        }
        for (int j=0; j < numOfStars; j++) {
          cout << "*";
        }
        numOfStars -= 2;
        cout << endl;
      }
      getch();
    }
    

};

int main() {
    //cout << "Hello, C++";
    Shapes myShapes;
    myShapes.displayTriangle();
    myShapes.displayReversedTriangle();
    return 0;
}

